src/goose/specs.clj

```diff
@@ -153,6 +158,7 @@
         (catch Exception _ false)))
 (s/def ::instant #(instance? Instant %))
 (s/def ::client-opts (s/keys :req-un [::broker ::queue ::retry-opts]))
+(s/def ::batch-args (s/and sequential? #(every? sequential? %) (s/* ::args-serializable?)))
```

Comment:
[nitpick] Can probably move `batch-args` to near `batch-opts` & name the section just Batch...

Reply:
>Are we keeping it because every? will raise an exception if a non-sequential arg is passed to it?

Yes, so that we're more explicit about it in the spec error as opposed to throwing an exception that mentions an `ISeq` was expected. 
We can get rid of it if that doesn't add much value.

Reply:
>Can probably move batch-args to near batch-opts & name the section just Batch...

Made this change